src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,54 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] arrayFrom = getIntArray(dateFrom.split("\\D"));
+        int[] arrayTo = getIntArray(dateTo.split("\\D"));
+        StringBuilder sb = new StringBuilder("Report for period ");
+        sb.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+
+        for (int i = 0; i < names.length; i++) {
+            sb.append(names[i]).append(" - ");
+            int sumName = 0;
+            for (int j = 0; j < data.length; j++) {
+
+                String[] words = data[j].split(names[i]);
```

Comment:
```suggestion
                String[] splitedData = data[j].split(" ");
                String name = splitedData[USER_NAME_INDEX];
```

I think it is better to split all data for one time and just get needed by index